### PR TITLE
Dart 3 migration

### LINKED ISSUE #1931: Firebase Upgrade Blocker / Google Play failed to upload artefacts
Time Tracking: MA-78

Summary: Error when uploading build to Google Play after `firebase_analytics` version upgrade.
>Google Play failed to upload artefacts. This release includes the com.google.android.gms.permission.AD_ID permission but your declaration on Play Console says your app doesn't use advertising ID. You must update your advertising ID declaration.

This issue is blocking Flutter and Firebase upgrades:
- Unable to upgrade Flutter on current version of Firebase
- Unable to upgrade Firebase due to AD_ID requirement

This is done when:
- [x] Firebase analytics alternatives documented
- [x] Document implications of adding AD_ID permission to Android and IOS [Collab](https://ucsdcollab.atlassian.net/wiki/spaces/WTS/pages/1575911587/Firebase+AI_ID+Permission+Workaround)

- [x] AD_ID permission workaround implemented for Android
- [x] PR 2077 Testing PASS on Android

- [x] Upgrade package firebase_core
- [x] Upgrade package firebase_crashlytics
- [x] Upgrade package firebase_messaging
- [x] Upgrade package firebase_analytics

- [x] CI/CD build errors resolved on iOS
- [x] CI/CD build errors resolved on Android

---
**Android Test Plan**:
- [PR 2080 Regression Test Plan](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7B46B62A53-2660-40B8-BB90-EB1DB78A2DDD%7D&file=Regression-Test-Plan-PR-2080-7.28.0-Android-QA.xlsx&action=default&mobileredirect=true)

**Android Status**:
- Testing failed with one issue.

**Android TODO**:
- [x] Fix title bar text in dark mode and re-test
- [x] Regression test Scanner on build 3198 (c3bryant)

---
**iOS Test Plan**:
- [PR 2080 Regression Test Plan](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7B4E4C5A15-9597-4FBC-AE66-70544ADB188F%7D&file=Regression-Test-Plan-PR-2080-7.28.0-iOS-QA.xlsx&action=default&mobileredirect=true)

**iOS Status**:
- Testing failed with one issue.

**iOS TODO**:
- [x] Fix title bar text in dark mode and re-test (Same issue as Android)

### PR DESCRIPTION
**Pull request by morebytes**
## Summary
Upgrades Campus Mobile repo to latest version of Flutter and Dart. 

## Changelog

- Upgrades to Flutter 3.24 and Dart 3.5 
- Upgrades Firebase packages to latest versions
- Upgrades any other necessary packages

## Test Plan
[PR-2080-Test-Plan-7.29.0-QA-iOS.xlsx](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7B9FC8CFB9-30F2-4179-9CFB-28020A983182%7D&file=[PR-2080-Test-Plan-7.29.0-QA-3283.xlsx](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7B9FC8CFB9-30F2-4179-9CFB-28020A983182%7D&file=PR-2080-Test-Plan-7.29.0-QA-3283.xlsx&action=default&mobileredirect=true)&action=default&mobileredirect=true)
[PR-2080-Test-Plan-7.29.0-QA-Android.xlsx](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7BCAA89DC5-07B5-4BDD-B657-68A8666C3670%7D&file=[PR-2080-Test-Plan-7.29.0-QA-3284.xlsx](https://ucsdcloud.sharepoint.com/:x:/r/sites/WorkplaceTechnologyServices-CampusMobileBuilds/_layouts/15/Doc.aspx?sourcedoc=%7BCAA89DC5-07B5-4BDD-B657-68A8666C3670%7D&file=PR-2080-Test-Plan-7.29.0-QA-3284.xlsx&action=default&mobileredirect=true)&action=default&mobileredirect=true)

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Uncomment this line to define a global platform for your project
-platform :ios, '11.0'
+platform :ios, '12.0'
```
Comment (c3bryant) [inline ios/Podfile]: Is this a required change?

File: lib/ui/triton_media/triton_media_detail_view.dart [hunk 2]
```diff
@@ -418,7 +418,7 @@ class SeekBarState extends State<SeekBar> {
                       .firstMatch("$_remaining")
                       ?.group(1) ??
                   '$_remaining',
-              style: Theme.of(context).textTheme.caption),
+              style: Theme.of(context).textTheme.bodySmall),
```
Comment (c3bryant) [inline lib/ui/triton_media/triton_media_detail_view.dart]: Is this equivalent?

File: pubspec.yaml [hunk 3]
```diff
@@ -3,40 +3,34 @@ description: Mobile app platform for campus.
...
     git:
       url: https://github.com/UCSD/flutter_dotenv.git
       ref: master
-  flutter_scandit_plugin:
-    git:
-      url: https://github.com/UCSD/flutter_scandit.git
-      ref: scandit-6.6.3
-  flutter_linkify: 5.0.2
-  flutter_local_notifications: 9.9.1
+  flutter_scandit_plugin: ^3.0.0
```
Comment (c3bryant) [inline pubspec.yaml]: We may want to update our fork if there is a chance this upstream will go away.

File: lib/ui/map/map_search_view.dart [hunk 4]
```diff
@@ -3,7 +3,7 @@ import 'package:campus_mobile_experimental/ui/common/container_view.dart';
 import 'package:campus_mobile_experimental/ui/map/quick_search_icons.dart';
 import 'package:campus_mobile_experimental/ui/map/search_bar.dart';
 import 'package:campus_mobile_experimental/ui/map/search_history_list.dart';
-import 'package:flutter/material.dart';
+import 'package:flutter/material.dart' hide SearchBar;
```
Comment (c3bryant) [inline lib/ui/map/map_search_view.dart:6]: What is the significance of this?

Comment (morebytes) [reply lib/ui/map/map_search_view.dart:6]: Resolves a naming conflict due to Flutter breaking changes

File: lib/ui/triton_media/triton_media_detail_view.dart (hunk 2 above)
Comment (morebytes) [reply lib/ui/triton_media/triton_media_detail_view.dart]: Yes

File: ios/Podfile (hunk 1 above)
Comment (morebytes) [reply ios/Podfile]: Minimum iOS version for latest Flutter is 12 [official doc](https://docs.flutter.dev/deployment/ios)

Comment (morebytes) [reply ios/Podfile]: Most older iPhones run 12+ iOS

File: pubspec.yaml (hunk 3 above)
Comment (morebytes) [reply pubspec.yaml]: Agreed, and also update our fork specifically to target Flutter 3.24 and latest supported iOS/Android versions if possible

**Comment by morebytes** [comment]
Thank you @klortiz13 for fixing the dark mode text bug. Now that it's out of the way, we are nearing completion.

**Review by morebytes — COMMENTED** [review]
Thoroughly tested, seems to be working. Will upgrade CodeMagic build pipeline after merging this

**Review by c3bryant — APPROVED** [review]

**Pull request merged**